### PR TITLE
Upgrade celery to 5.0.4

### PR DESCRIPTION
**Pull request by iamsobanjaved**
**Description:**  
- Upgrade celery to 5.0.4
- Bumped Version
- Updated python classifier as python 3.5 is already dropped

**JIRA:** Link to JIRA ticket

**Merge checklist:**
- [x] All reviewers approved
- [x] CI build is green
- [x] Version bumped
- [x] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [x] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

**Review by mzulqarnain1 — APPROVED** [review]

File: CHANGELOG.rst [hunk 1]
```diff
@@ -15,6 +15,10 @@ Unreleased
 ~~~~~~~~~~
 *
 
+[0.8.5] - 2020-12-24
+~~~~~~~~~~~~~~~~~~~~~
+* Upgrading celery to 5.0.4
+
```
Comment (nedbat) [inline CHANGELOG.rst:21]: Does this need to be deployed?  Generally you don't need a new version number just to upgrade a dependency.

Comment (iamsobanjaved) [reply CHANGELOG.rst:21]: We need to release the package to update python classifiers on PyPI as python 3.5 support dropped earlier.

File: .travis.yml [hunk 2]
```diff
@@ -3,7 +3,6 @@ dist: xenial
 python:
   - 3.8
 env:
-  - TOXENV=django22-celery44
```
Comment (jmbowman) [inline .travis.yml:6]: Should we hold off on removing this testing until edx-platform has upgraded to celery 5.x?

File: requirements/celery50.txt [hunk 3]
```diff
@@ -1,9 +1,9 @@
 amqp==5.0.2               # via kombu
 billiard==3.6.3.0         # via celery
-celery==5.0.5             # via edx-celeryutils
+celery==5.0.4             # via -c requirements/constraints.txt, edx-celeryutils
```
Comment (jmbowman) [inline requirements/celery50.txt:3]: Was there any reason to downgrade this?

File: requirements/constraints.txt [hunk 4]
```diff
@@ -27,5 +27,5 @@ twine<2.0
 # greater versions drops support for python 3.5
 stevedore<2.0
 
-# Keeping celery 4.4.7 in the production currently
-celery<5.0
+# pinning celery to latest release
+celery==5.0.4
```
Comment (jmbowman) [inline requirements/constraints.txt:31]: This should probably be `celery<6.0` instead, to allow point release upgrades to keep progressing.

**Pull request merged**